src/CppUTest/CommandLineTestRunner.cpp

```diff
@@ -145,8 +153,11 @@ bool CommandLineTestRunner::parseArguments(TestPlugin* plugin)
     return false;
   }
 
-  if (arguments_->isJUnitOutput())
+  if (arguments_->isJUnitOutput()) {
     output_= createJUnitOutput(arguments_->getPackageName());
+    if (arguments_->isVerbose())
+      output_ = createCompositeOutput(output_, createConsoleOutput());
+  }
   else
     output_ = createConsoleOutput();
   return true;
```

Comment:
You are right about that JUnitOutput shouldn't know about it, which is why I prefer your solution, too.

I just wish we could do the test runner side more elegantly....